Adding FVFunctorNeumannBC and FVFunctorConvectiveHeatFluxBC (#21668)
idaholab/moose: modules/heat_conduction/doc/content/source/fvbcs/FVFunctorConvectiveHeatFluxBC.md

# FVFunctorConvectiveHeatFluxBC

!syntax description /FVBCs/FVFunctorConvectiveHeatFluxBC

# Description

> This boundary condition computes convective heat flux $q'' = H \cdot (T - T_{inf})$, where $H$ is convective heat transfer coefficient,

Reviewer: I think we refer to it as lower case `h` usually....being nit-picky

> This boundary condition computes convective heat flux $q'' = h \cdot (T - T_{bulk})$, where $H$ is convective heat transfer coefficient,

Reviewer 2: ```suggestion
This boundary condition computes convective heat flux $q'' = h \cdot (T - T_{bulk})$, where $h$ is the convective heat transfer coefficient,
```

> $T$ is the temperature, and $T_{bulk}$ is far field temperature. Both $H$ and $T_{bulk}$ are coupled as functors.

Reviewer 2: ```suggestion
$T$ is the temperature, and $T_{bulk}$ is the far-field temperature. Both $h$ and $T_{bulk}$ are functors, which enables various spatial, variable and other dependences.
```

> The domain can be specified as either a fluid or a solid using $is_solid$. For a solid domain, the equation above is applied.

Reviewer: The domain of the `variable` can be specified ...(i.e. is the variable the solid or the fluid temperature).

For a fluid domain, the negative of the heat flux is applied. This allows for easier implementation of a double Robin
boundary condition.

> See [CoupledConvectiveHeatFluxBC](CoupledConvectiveHeatFluxBC.md) for a similar boundary condition coupled to variables, and see [FunctorThermalResistanceBC](FunctorThermalResistanceBC.md) for a combined conduction, convection, and radiative boundary condition

Reviewer 2: ```suggestion
Similar objects:
- [CoupledConvectiveHeatFluxBC](CoupledConvectiveHeatFluxBC.md) for a similar boundary condition coupled to variables, for finite elements
- [FunctorThermalResistanceBC](FunctorThermalResistanceBC.md) for a combined conduction, convection, and radiative boundary condition, with a constant outside ambient temperature
```

>  with functor material properties.

Reviewer 2: ```suggestion
```

# FVFunctorConvectiveHeatFluxBC

!syntax description /FVBCs/FVFunctorConvectiveHeatFluxBC

# Description

This boundary condition computes convective heat flux $q'' = h \cdot (T - T_{bulk})$, where $h$ is the convective heat transfer coefficient,
$T$ is the temperature, and $T_{bulk}$ is the far-field temperature. Both $h$ and $T_{bulk}$ are functors, which enables various spatial, variable and other dependences.

> The domain of the variable can be specified as either a fluid or a solid using $is\_solid$. For a solid domain, the equation above is applied.

Reviewer 3: ```suggestion
The domain of the variable can be specified as either a fluid or a solid using the `is_solid` parameter. For a solid domain, the equation above is applied.
```
